cmake/InstallDirs.cmake

```diff
@@ -0,0 +1,52 @@
+if(WIN32)
+  if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+    set(program_files "$ENV{ProgramFiles}")
+
+    # For 32 bit builds.
+    if(CMAKE_C_SIZEOF_DATA_PTR EQUAL 4 AND ENV{ProgramFiles\(x86\)})
+      set(program_files "$ENV{ProgramFiles\(x86\)}")
+    endif()
+
+    if(NOT program_files)
+      if(NOT CMAKE_C_SIZEOF_DATA_PTR EQUAL 4)
```

Comment:
I'm not a cmake expert but shouldn't actually `CMAKE_SIZEOF_VOID_P` better be used here? Because it's at least [documented](https://cmake.org/cmake/help/latest/variable/CMAKE_SIZEOF_VOID_P.html).

Actually I found the [issue](https://gitlab.kitware.com/cmake/cmake/-/issues/18312) in cmake tracker discussing if `Program Files` is even correct default anymore but it's 3 years old and no changes have been made it seems.

Reply:
The reason I did this was that it was always setting `C:\Program Files (x86)` as the prefix even for 64 bit builds. I think this should be more reasonable and in the future we can generate an `.msi` with CPack or something.

Reply:
Oh right, that was the issue you linked, never mind.